### ISSUE TITLE
`object-property-newline`: does not apply to deconstructed objects

### ISSUE THREAD

**Issue by longzheng** [eslint-stylistic/eslint-stylistic#628]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

I expected the [`object-property-newline`](https://eslint.style/rules/default/object-property-newline#object-property-newline) rule to also handle a deconstructed object, not just the type.

For example in the following examples, `object-property-newline` correctly puts the properties of the type on new lines, but not the values

```ts
function test({
  value1, value2, value3,
}: {
  value1: string;
  value2: number;
  value3: boolean;
}) {}
```

```ts
const {
  value1, value2, value3,
} = {} as {
  value1: string;
  value2: number;
  value3: boolean;
};
```

I expected it to be
```ts
function test({
  value1,
  value2,
  value3,
}: {
  value1: string;
  value2: number;
  value3: boolean;
}) {}
```

```ts
const {
  value1, 
  value2, 
  value3,
} = {} as {
  value1: string;
  value2: number;
  value3: boolean;
};
```

There was some discussions about a[ `destructuring-property-newline` rule on the ESLint repo](https://github.com/eslint/eslint/issues/9259) but it hit a bit of a wall and got archived. 

### Reproduction

https://stackblitz.com/edit/github-fgybbr?file=eslint.config.js,src%2Findex.ts

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
